proposals/0072-feature-gate-threshold-automation.md

```diff
@@ -0,0 +1,247 @@
+2. The runtime automatically activates the feature on the next epoch boundary
+
+The key-holder is the one who (with the help of the solana-cli) assesses the
+amount of stake that recognizes the feature and decides whether it is safe to
+activate. This is obviously brittle and subject to human error.
+
+This SIMD proposes that the runtime itself replaces the human to assess the
+amount of stake that supports a particular queued feature and only
+automatically activates it when a preset threshold is met. 
+
+## New Terminology
+
+- Feature Gate program: The BPF program that will own all feature accounts.
+- Feature-Request account: The PDA under the Feature Gate program used to track
+features requested for activation.
+- Feature-Queue account: The PDA under the Feature Gate program used to track
+features queued for activation that must have support signaled by nodes.
+
+## Detailed Design
+
+Before this SIMD is published, we will deploy a BPF program to manage feature
+activations, which will allow us to revoke pending feature activations
+(previously not possible).
+The feature activation process will otherwise be unchanged, so no SIMD is
+required.
+This program will be upgradeable; the authority and process through which it
+will be upgraded is outside the scope of this SIMD. Since this SIMD requires an
+update to the program, the upgrade details will be determined in parallel.
+
+The initial Feature Gate Program can be found in this pull request to the
+Solana Program Library:
+<https://github.com/solana-labs/solana-program-library/pull/4987>.
+The runtime changes and feature activation that will enable this program can be
+found in this feature tracking issue:
+<https://github.com/solana-labs/solana/issues/33547>.
+
+### Overview
+
+Consider two arbitrary versions of the Solana runtime, where the newer version
+has more features in its feature set.
+
+| v1.14.25  | v1.16.0   |
+| --------- | --------- |
+| Feature A | Feature A |
+|           | Feature B |
+| Feature C | Feature C |
+|           | Feature D |
+
+The runtime should activate each of these features when a strong majority of
+the network supports it.
+"Strong majority" in this context is defined by the current feature activation
+process as 95% of stake.
+“Support” for a feature means that a node is running a version of the Solana
+runtime where the code required for the feature is present. For instance, on a
+cluster with 50% of stake running v1.14.25 and 50% of stake running v1.16.0,
+only Features A and C should be eligible for activation.
+
+This SIMD proposes 3 steps over the span of one epoch to determine which
+feature gates are queued for activation and assess stake support:
+
+1. On an epoch boundary, the runtime generates a list of feature-gates queued
+for activation.
+2. Validators signal which of the queued feature-gates they support.
+3. On the next epoch boundary, the runtime activates the feature-gates that
```

Comment:
This should be removed from the proposal if the proposal's goal is detection of features

Reply:
The goal of the proposal is to replace what is currently a human saying "yeah, I think enough of the validators recognize this feature that activation won't cause a partition in the network or piss anyone off too badly" with an automated process in the runtime. So yes, as with the current system, it ends with feature activation.

It is what you termed "Version Availability". If there is a substantive reason why this Version Availability check needs to happen before whatever governance system is adopted, instead of after as we've proposed, please explain.